### ISSUE TITLE
Add support for an explicit srcjar for scala_import

### ISSUE THREAD

**Issue by dflemstr** [bazelbuild/rules_scala#486]
It would be nice to have `scala_import` be on par with [`java_import`](https://docs.bazel.build/versions/master/be/java.html#java_import) in this regard, especially since `scala_import_external` suggests it should be supported.

I could probably take on fixing this if somebody could mentor me regarding a preferred solution.

**Comment by ittaiz** [comment]
Hi David,
Thanks for reaching out!
I think we’re in the midst of a few changes (see move to new providers,
refactoring of the whole files and in general discussion of new JavaInfo).
I think that in a few weeks the dust will settle and we’ll tackle this. How
does that sound?

**Comment by dflemstr** [comment]
Sounds good, I mostly wanted to create this issue to have something to track progress, it's not very urgent at all.  If I happen to find the time to fix it on top of the old `scala_import` implementation on my own, would you accept that changeset anyway or still prefer to not have too many moving parts while the refactoring is going on?

**Comment by jjudd** [comment]
We added `srcjar` support for Scala import in this commit: https://github.com/andyscott/rules_scala/pull/2/commits/a457c06e71f6fa89379ef96db4608ec60df8e547, which is currently PR to this PR: https://github.com/bazelbuild/rules_scala/pull/477

We currently use it with `scala_import_external`.

I'm not totally certain the status of that PR. We're busy rolling out Bazel for Scala to all of engineering internally over the next two weeks and have a bunch of forks that we'll need to clean up and get back onto master once we've migrated.

**Comment by ittaiz** [comment]
Right, that's part of the turmoil I was referring too since I remembered
discussion about it but didn't remember it taking place on a PR.
That PR is currently stuck on intellij not actually using the srcjar so the
plan IIRC is to fix intellij plugin and then move forward with it

**Comment by johnynek** [comment]
This dependency knot is a pain. I am losing state. I think bazel has to be updated for intellij to not lose state using only JavaInfo. So, that has to land, then we have to send the PR to the plugin, then we have to update locally.

@jjudd says we don't see a regression if we merge now. Can anyone confirm that? @andyscott  @gkk-stripe ?

If we merge and regress on intellij it will be an issue for us at Stripe, and probably others.

**Comment by jjudd** [comment]
@johnynek agreed. It's spaghetti right now. 

@johnynek We'll need this https://github.com/andyscott/rules_scala/commit/a457c06e71f6fa89379ef96db4608ec60df8e547 merged into @andyscott's branch before we merge if we want IntelliJ to continue working.

**Comment by andyscott** [comment]
See https://github.com/bazelbuild/rules_scala/pull/487 for the sum of all lines of development.
